frontends/web/src/routes/buy/info.tsx

```diff
@@ -34,45 +35,50 @@ type TProps = {
 
 export const BuyInfo = ({ code, accounts }: TProps) => {
   const [selected, setSelected] = useState<string>(code);
-  const [options, setOptions] = useState<TOption[]>();
+  const [groupedOptions, setGroupedOptions] = useState<TGroupedOption[]>();
   const [disabled, setDisabled] = useState<boolean>(false);
 
   const { t } = useTranslation();
 
   const checkSupportedCoins = useCallback(async () => {
     try {
       const supportedAccounts = await getExchangeSupportedAccounts(accounts);
-      const options =
-        supportedAccounts.map(({ name, code, coinCode, bitsuranceStatus }) => ({ label: `${name}`, value: code, coinCode, disabled: false, insured: bitsuranceStatus === 'active' }));
-      setOptions(await setOptionBalances(options));
+      const accountNames = getAccountsByKeystore(supportedAccounts);
+      const groupedOpts: TGroupedOption[] = accountNames.map(grouped => ({
+        label: `${grouped.keystore.name} ${isAmbiguiousName(grouped.keystore.name, accountNames) ? `(${grouped.keystore.rootFingerprint})` : ''}`,
+        connected: grouped.keystore.connected,
+        options: grouped.accounts.map((account) => ({ label: account.name, value: account.code, coinCode: account.coinCode, disabled: false, insured: account.bitsuranceStatus === 'active' })) as TOption[]
+      }));
+      setGroupedOptions(groupedOpts);
+      getBalancesForGroupedAccountSelector(groupedOpts)
+        .then(options => {
+          setGroupedOptions(options);
+        });
     } catch (e) {
       console.error(e);
     }
 
   }, [accounts]);
 
+  useEffect(() => {
+    checkSupportedCoins();
+  }, [checkSupportedCoins]);
+
+
   const maybeProceed = useCallback(async () => {
```

Comment:
I wanted to refactor the GroupedAccountSelector to just accept an IAccount object and do all the object / group constructing in that component. 

However, we need the group options outside of that component sometimes like here.

I think it still can be refactored though, I could work on this next. My idea is so that we can just pass an `IAccount[]` and "it just works" (automatically grouped by keystore like we have) without having to do all the grouping in the parent component like currently.